lib/seattleflu/id3c/cli/command/etl/redcap_det_fh_airs.py

```diff
@@ -732,109 +816,355 @@ def airs_build_location_info(db: DatabaseSession, cache: TTLCache,
+        'stomach_pain':     '271681002',
+        'vomiting':         '422400008',
+    }
 
-            specimen_observation_entry_v2 = create_specimen_observation_entry(
-                specimen_reference = specimen_reference_v2,
-                patient_reference = patient_reference,
-                encounter_reference = initial_encounter_reference)
-        else:
-            LOG.debug(f"No identifier found for v2 barcode «{specimen_barcode_v2}»")
-            LOG.info("No identifier found for barcode. Creating v2 encounter for record instance without sample")
+    if symptom_name.lower() in airs_symptoms_to_snomed_map:
+        snomed_code = airs_symptoms_to_snomed_map[symptom_name.lower()]
+        return {
+            'system': SNOMED_CT_SYSTEM,
+            'code': snomed_code,
+            'display': snomed_symptoms[snomed_code]
+        }
     else:
-        LOG.debug("Creating v2 encounter for record instance without sample")
+        raise UnknownSymptomNameError(f"Unknown symptom name «{symptom_name}»")
 
-    # Do not log warning for unrecognized barcodes, which are being tracked outside of ID3C.
-    if specimen_received_v2 and specimen_identifier_v2 and not specimen_entry_v2:
-        LOG.warning("Skipping v2 record instance. We think the specimen was received, "
-             "but we're unable to create the specimen_entry for record: "
-             f"{redcap_record_instance.get('record_id')}, instance: {redcap_record_instance.get('redcap_repeat_instance')}"
-             )
 
-    return (specimen_entry, specimen_observation_entry, specimen_entry_v2, specimen_observation_entry_v2)
+def airs_build_contained_and_diagnosis(patient_reference: dict, record: REDCapRecord,
+        encounter_type: EncounterType, system_identifier: str) -> Tuple[list, list]:
+
+    def map_severity_to_snomed(severity: str) -> Optional[dict]:
+        snomed_severity_map = {
+            '1':    {'system': SNOMED_CT_SYSTEM, 'code': '255604002', 'display': 'mild'},
+            '2':    {'system': SNOMED_CT_SYSTEM, 'code': '6736007', 'display': 'moderate'},
+            '3':    {'system': SNOMED_CT_SYSTEM, 'code': '24484000', 'display': 'severe'},
+        }
+
+        if severity in snomed_severity_map:
+            return snomed_severity_map[severity]
+        else:
+            return None
+
+    def build_condition(patient_reference: dict, symptom_name: str, symptom_severity: str,
+        onset_date: str, system_identifier: str) -> Optional[dict]:
+        """ Returns a FHIR Condition resource. """
+        mapped_symptom_code = map_symptom_to_snomed(symptom_name)
+        mapped_severity_code = map_severity_to_snomed(symptom_severity)
+
+        if not mapped_symptom_code or not mapped_severity_code:
+            return None
+
+        # XXX TODO: Define this as a TypedDict when we upgrade from Python 3.6 to
+        # 3.8.  Until then, there's no reasonable way to type this data structure
+        # better than Any.
+        #   -trs, 24 Oct 2019
```

Comment:
I'm guessing this got copied from elsewhere? :) We have already made that upgrade now - though making that update may be out of scope here

Reply:
Yes, this comment got copied from one of the other ETLs (it shows up in multiple), so I think we should address it in a separate PR.